src/plots/plots.js

```diff
@@ -1291,9 +1291,16 @@ plots.supplyTraceDefaults = function(traceIn, traceOut, colorIndex, layout, trac
         coerce('meta');
 
         if(Registry.traceIs(traceOut, 'showLegend')) {
-            traceOut._dfltShowLegend = true;
-            coerce('showlegend');
+            Lib.coerce(traceIn, traceOut,
+                _module.attributes.showlegend ?
+                _module.attributes :
+                plots.attributes,
+                'showlegend'
+            );
```

Comment:
That's a very clean workaround. Nice job!

Alternatively, we could have moved the `coerce('showlegend'`) call to the trace module defaults, but then the `coerce('legendgroup')` call and `_dfltShowLegend` logic would be pretty annoying to get right. Your way is much better.

Reply:
One small comment. I would prefer having:

```js
            Lib.coerce(traceIn, traceOut,
                _module.attributes.showlegend ? _module.attributes :+1:  plots.attributes,
                'showlegend'
            );
```

so that new lines in the `Lib.coerce` call are reserved for its arguments not ternary-operator results.

Reply:
Or something like:

```js
            Lib.coerce(traceIn, traceOut,
                _module.attributes.showlegend ? 
                   _module.attributes : 
                   plots.attributes,
                'showlegend'
            );
```

would be ok too.

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/4386/commits/a9759a27029ce39dd1071d570b8254d0ea667d00.